### PR TITLE
Added docstrings for the majority of public calls

### PR DESCRIPTION
**Pull request by CraigLn**
This should bring the codebase inline with proposed SDK doc website changes

### CODE REVIEW

File: Sources/PubNub/Events/Subscription/SubscriptionStream.swift [hunk 1]
```diff
@@ -27,10 +27,14 @@
 
 import Foundation
 
+/// A channel or group that has successfully beend subscribed or unsubscribed
```
Comment (davidnub) [inline Sources/PubNub/Events/Subscription/SubscriptionStream.swift]: Typo: `beend` -> `been`

File: Sources/PubNub/Helpers/AnyJSON.swift [hunk 2]
```diff
@@ -97,9 +99,13 @@ extension AnyJSON: Codable {
 
 // MARK: - Collection
 
+/// The index of for an underlying collection
```
Comment (davidnub) [inline Sources/PubNub/Helpers/AnyJSON.swift]: Grammar change: `of for` -> `for`

File: Sources/PubNub/Helpers/AnyJSON.swift [hunk 3]
```diff
@@ -351,6 +341,7 @@ extension AnyJSON: ExpressibleByStringLiteral {
 
 // MARK: - AnyJSONError
 
+/// An `Error` that occurred as a result of perorming an action using an `AnyJSON`
```
Comment (davidnub) [inline Sources/PubNub/Helpers/AnyJSON.swift]: Typo: `perorming` -> `performing`

File: Sources/PubNub/Helpers/Crypto.swift [hunk 4]
```diff
@@ -64,26 +68,31 @@ public struct Crypto: Hashable {
       }
     }
 
+    /// Block size for the algoritm
```
Comment (davidnub) [inline Sources/PubNub/Helpers/Crypto.swift]: Typo: `algoritm` -> `algorithm`

File: Sources/PubNub/Helpers/Crypto.swift [hunk 5]
```diff
@@ -64,26 +68,31 @@ public struct Crypto: Hashable {
...
     public var blockSize: Int {
       switch self {
       case .aes:
         return kCCBlockSizeAES128
       }
     }
 
+    /// Key size for the algoritm
```
Comment (davidnub) [inline Sources/PubNub/Helpers/Crypto.swift]: Typo: `algoritm` -> `algorithm`

File: Sources/PubNub/Helpers/JSONCodable.swift [hunk 6]
```diff
@@ -29,8 +29,11 @@ import Foundation
 
 // MARK: JSONCodable
 
-public protocol JSONCodable {
+/// A object that can be represented as JSON
```
Comment (davidnub) [inline Sources/PubNub/Helpers/JSONCodable.swift]: Grammar change: `A object` -> `An object`

File: Sources/PubNub/Helpers/WeakBox.swift [hunk 7]
```diff
@@ -27,19 +27,21 @@
 
 import Foundation
 
+/// A container that stores a`weak` reference to its `Element`
```
Comment (davidnub) [inline Sources/PubNub/Helpers/WeakBox.swift]: Grammar change: ``a`weak``` -> ``a `weak```

File: Sources/PubNub/Networking/Convertibles+PubNub.swift [hunk 8]
```diff
@@ -27,19 +27,15 @@
 
 import Foundation
 
+/// An object that is capable of converting into a URL
```
Comment (davidnub) [inline Sources/PubNub/Networking/Convertibles+PubNub.swift]: Grammar change: `a URL` -> `an URL`

File: Sources/PubNub/Networking/Convertibles+PubNub.swift [hunk 9]
```diff
@@ -27,19 +27,15 @@
...
 public protocol URLConvertible {
+  /// The converted `URL` or the error that occurred during converting
   var asURL: Result<URL, Error> { get }
 }
 
-enum URLConvertibleError: Error {
+/// An `Error` that occured when generating a `URL` from a `URLConvertible`
```
Comment (davidnub) [inline Sources/PubNub/Networking/Convertibles+PubNub.swift]: Grammar change: ``a `URL` `` -> ``an `URL` ``
Also: ``a `URLConvertible` `` -> ``an `URLConvertible` ``

File: Sources/PubNub/Networking/Replaceables+PubNub.swift [hunk 10]
```diff
@@ -29,27 +29,68 @@ import Foundation
 
 // MARK: - URLSession
 
+/// An object capable of replacing a `URLSession`
```
Comment (davidnub) [inline Sources/PubNub/Networking/Replaceables+PubNub.swift]: Grammar change: ``a `URLSession` ``

File: Sources/PubNub/Networking/Replaceables+PubNub.swift [hunk 11]
```diff
@@ -29,27 +29,68 @@ import Foundation
...
 public protocol URLSessionReplaceable {
+  /// Creates a session with the specified session configuration, delegate, and operation queue.
   init(configuration: URLSessionConfiguration, delegate: URLSessionDelegate?, delegateQueue: OperationQueue?)
-
+  /// An app-defined descriptive label for the session.
   var sessionDescription: String? { get set }
+  /// The operation queue provided when this object was created.
   var delegateQueue: OperationQueue { get }
+  /// A configuration object that defines behavior and policies for a URL session.
```
Comment (davidnub) [inline Sources/PubNub/Networking/Replaceables+PubNub.swift]: Grammar change: `a URL` -> `an URL`

File: Sources/PubNub/Networking/Replaceables+PubNub.swift [hunk 12]
```diff
@@ -29,27 +29,68 @@ import Foundation
...
   var configuration: URLSessionConfiguration { get }
-
-  func dataTask(with: URLRequest) -> URLSessionDataTask
+  /// Creates a task that retrieves the contents of a URL based on the specified URL request object.
```
Comment (davidnub) [inline Sources/PubNub/Networking/Replaceables+PubNub.swift]: Grammar change: `a URL` -> `an URL`

File: Sources/PubNub/Networking/Replaceables+PubNub.swift [hunk 13]
```diff
@@ -29,27 +29,68 @@ import Foundation
...
+  ///
+  /// By creating a task based on a request object, you can tune various aspects of the task’s behavior,
+  /// including the cache policy and timeout interval.
+  ///
+  /// After you create the task, you must start it by calling its resume() method.
+  ///
+  /// - Parameter with: A URL request object that provides request-specific information such as the URL,
```
Comment (davidnub) [inline Sources/PubNub/Networking/Replaceables+PubNub.swift]: Grammar change: `A URL` -> `An URL`

File: Sources/PubNub/Networking/Request/Operators/AutomaticRetry.swift [hunk 14]
```diff
@@ -27,31 +27,55 @@
...
 public struct AutomaticRetry: RequestOperator, Hashable {
+  /// Exponential backoff twice for any 500 response code or `URLError` contained in `defaultRetryableURLErrorCodes`
   public static var `default` = AutomaticRetry()
-  public static var none = AutomaticRetry(policy: .none)
+  /// No retry will be performed
+  public static var none = AutomaticRetry(retryLimit: 1)
+  /// Retry immediately twice on lost network connection
   public static var connectionLost = AutomaticRetry(policy: .immediately,
                                                     retryableURLErrorCodes: [.networkConnectionLost])
+  /// Exponential backoff twice twice when not internet connection is detected
```
Comment (davidnub) [inline Sources/PubNub/Networking/Request/Operators/AutomaticRetry.swift]: Typo: Extra `twice` remove one
Also grammar change: `not` -> `no`

File: Sources/PubNub/Networking/Router.swift [hunk 15]
```diff
@@ -48,17 +48,20 @@ public protocol RouterConfiguration {
 }
 
 extension RouterConfiguration {
+  /// The scheme used when creating the URL for the request
   public var urlScheme: String {
     return useSecureConnections ? "https" : "http"
   }
 
+  /// True if the subscribeKey exsits and is not an empty `String`
```
Comment (davidnub) [inline Sources/PubNub/Networking/Router.swift]: Typo: `exsits` -> `exists`

File: Sources/PubNub/Networking/Router.swift [hunk 16]
```diff
@@ -48,17 +48,20 @@ public protocol RouterConfiguration {
...
   public var subscribeKeyExists: Bool {
     guard let subscribeKey = subscribeKey, !subscribeKey.isEmpty else {
       return false
     }
     return true
   }
 
+  /// True if the publishKey exsits and is not an empty `String`
```
Comment (davidnub) [inline Sources/PubNub/Networking/Router.swift]: Typo: `exsits` -> `exists`

File: Sources/PubNub/Networking/Router.swift [hunk 17]
```diff
@@ -108,81 +89,36 @@ public enum HTTPMethod: String {
 
 // MARK: - Router
 
+/// Collects together and assembles the separate pieces used to create a URLRequest
```
Comment (davidnub) [inline Sources/PubNub/Networking/Router.swift]: Grammar change: `a URLRequest` -> `an URLRequest`

File: Sources/PubNub/Subscription/SubscriptionState.swift [hunk 18]
```diff
@@ -27,19 +27,27 @@
 
 import Foundation
 
+/// State of a PubNub subscription lifecycle
 public struct SubscriptionState {
+  /// Connection status
   public var connectionState: ConnectionStatus = .disconnected
+  /// Dictionary the maps channel name to the Channel object
```
Comment (davidnub) [inline Sources/PubNub/Subscription/SubscriptionState.swift]: Grammar change: `the maps` -> `that maps`

File: Sources/PubNub/Subscription/SubscriptionState.swift [hunk 19]
```diff
@@ -27,19 +27,27 @@
...
   public var channels: [String: PubNubChannel] = [:]
+  /// Dictionary the maps group name to the group Channel object
```
Comment (davidnub) [inline Sources/PubNub/Subscription/SubscriptionState.swift]: Grammar change: `the maps` -> `that maps`

File: Sources/PubNub/Subscription/SubscriptionState.swift [hunk 20]
```diff
@@ -88,6 +101,9 @@ public struct SubscriptionState {
     return state
   }
 
+  /// Search for and update a channel name matching the id with the provided states/
```
Comment (davidnub) [inline Sources/PubNub/Subscription/SubscriptionState.swift]: Typo: Ending `/` probably doesn't need to be there?

File: Sources/PubNub/Events/Subscription/ObjectEvent.swift [hunk 21]
```diff
@@ -27,47 +27,69 @@
...
 public enum MembershipEvents {
+  /// The IDs of the User and Space whose membership was added
   case userAddedOnSpace(MembershipEvent)
+  /// The IDs of the User and Space whose membership was updated
   case userUpdatedOnSpace(MembershipEvent)
+  /// The IDs of the User and Space that have become separated
   case userDeletedFromSpace(MembershipIdentifiable)
 }
 
+/// A way to uniquely identify a Membership between a User and a Space
```
Comment (polarweasel) [inline Sources/PubNub/Events/Subscription/ObjectEvent.swift]:
```suggestion
/// Uniquely identifies a Membership between a User and a Space
```

File: README.md [hunk 22]
```diff
@@ -17,7 +17,7 @@
 ## Requirements
 
 -   iOS 8.0+ / macOS 10.10+ / tvOS 9.0+ / watchOS 2.0+
```
[the remark this answers is not in the document]
Comment (CraigLn) [reply README.md]: Updated to include `Mac Catalyst 13.0+`

File: Sources/PubNub/Errors/PubNubError.swift [hunk 23]
```diff
@@ -118,7 +119,13 @@ public struct PubNubError: Error {
...
     case certificateTrustFailure
 
-    // System Error Inbound
+    // Cancellation
+    case sessionDeinitialized
+    case sessionInvalidated
+    case clientCancelled
+    case longPollingRestart
+
+    // Response Receiving
```
[the remark this answers is not in the document]
Comment (CraigLn) [reply Sources/PubNub/Errors/PubNubError.swift]: Changed to `Response Received` to better match the domain type

File: Sources/PubNub/Events/EventStream.swift [hunk 24]
```diff
@@ -27,12 +27,15 @@
 
 import Foundation
 
-public protocol EventStream {
+/// An object capable of streaming a event of objects
```
[the remark this answers is not in the document]
Comment (CraigLn) [reply Sources/PubNub/Events/EventStream.swift]: Updated the wording `An object capable of broadcasting other objects as events`. Let me know if that is more clear

[the remark this answers is not in the document]
Comment (polarweasel) [reply Sources/PubNub/Events/EventStream.swift]: Yup, that's great. Thanks!

**Pull request merged**